### PR TITLE
Python Technical Interview Expert

### PR DESCRIPTION
**Pull request by simonayotte**

### CODE REVIEW

File: data/presets.ts [hunk 1]
```diff
@@ -85,6 +85,31 @@ Here are some rules:
     web_search: true,
     date: "2024-03-26",
   },
+  {
+    id: "python-technical-interview-expert",
+    name: "Python Technical Interview Expert",
+    instructions: `You are a Python Developer that provides expert-level insights and solutions.
+    You are an expert in data structures and algorithms.
```
Comment (samuelkraft) [inline data/presets.ts:92]: Please indent these lines to improve formatting

File: data/presets.ts [hunk 2]
```diff
@@ -85,6 +85,31 @@ Here are some rules:
...
+    Step 3: Optimization - From here, we should brainstorm possible ways to improve our brute force solution.
+    We might consider different algorithms, data structures, or computational techniques that could make our solution more efficient.
+    Step 4: Walk-through - Now, I'd like you to walk me through the optimized solution using our initial example.`,
+    description: "A technical interview expert in Python.",
+    icon: "python",
+    creativity: "low",
+    model: "openai-gpt-4-turbo",
+    web_search: true,
+    date: "2024-04-24",
+  },
```
Comment (samuelkraft) [inline data/presets.ts:112]: Add the `author` field with your name and url

**Review by samuelkraft — COMMENTED** [review]
Thanks for the contribution! 🙌 

I feel like this preset could benefit from being a bit more concise for it's use-case, right now the instructions and rules are mixed with steps that somewhat contradict each other. Did you test this for the use-case with good results?

**Comment by simonayotte** [comment]
> Thanks for the contribution! 🙌
> 
> I feel like this preset could benefit from being a bit more concise for it's use-case, right now the instructions and rules are mixed with steps that somewhat contradict each other. Did you test this for the use-case with good results?

I tested it with a couple of easy problems on Leetcode and the output seems great for the particular use case! The output can be a bit verbose depending on the problem but I think this is a better alternative than having to ask for more clarification. 
image

File: data/presets.ts [hunk 3]
```diff
@@ -85,6 +85,34 @@ Here are some rules:
     web_search: true,
     date: "2024-03-26",
   },
+  {
+    id: "python-technical-interview-expert",
+    name: "Python Technical Interview Expert",
```
Comment (samuelkraft) [inline data/presets.ts:90]:
```suggestion
    name: "Technical Interview Expert",
```

File: data/presets.ts [hunk 4]
```diff
@@ -85,6 +85,34 @@ Here are some rules:
...
+  - Prefer standard library functions and modules whenever possible, and limit use of third-party packages to those that are well-maintained and commonly used in the industry.
+  - Include links to reputable sources for further reading (when beneficial), prefer official documentation.
+  - For each problem I present to you, please guide me through the following stages:
+  Step 1: Understanding and Visualization - I'd like you to help me draw a representative example or diagram, to better understand the problem and its constraints.
+  Step 2: Brute Force Approach - Let's discuss a simple but potentially inefficient way to solve the problem. This will serve as our baseline strategy.
+  Step 3: Optimization - From here, we should brainstorm possible ways to improve our brute force solution.
+  We might consider different algorithms, data structures, or computational techniques that could make our solution more efficient in terms of time and space complexity.
+  Step 4: Walk-through - Now, I'd like you to walk me through the optimized solution using our initial example.`,
+    description: "A technical interview expert in Python.",
```
Comment (samuelkraft) [inline data/presets.ts:105]:
```suggestion
    description: "A Python Expert helping you through technical interview questions.",
```

File: data/presets.ts [hunk 5]
```diff
@@ -384,6 +412,39 @@ Here are the rules you must follow:
 ];
 
 const misc: Preset[] = [
+  {
+    id: "news-reader",
```
Comment (samuelkraft) [inline data/presets.ts:416]: This preset has been removed

**Review by samuelkraft — COMMENTED** [review]
Great! 

What do you think about these slight title/desc changes? A shorter title improves the display and still works thanks to the icon, and paired with a more detailed description.

**Comment by simonayotte** [comment]
> Great!
> 
> What do you think about these slight title/desc changes? A shorter title improves the display and still works thanks to the icon, and paired with a more detailed description.

Great! Looks good to me!

**Comment by samuelkraft** [comment]
Can you check your merge again please? The Quality Control preset got removed invorrectly.

Then ready to go :)

**Comment by simonayotte** [comment]
> Can you check your merge again please? The Quality Control preset got removed invorrectly.
> 
> Then ready to go :)

Oops. Sorry! Should be fixed now. 😅

**Pull request merged**